app/ts/inequality/StateSymbol.ts

```diff
@@ -0,0 +1,213 @@
+     */
+     get dockingPoint(): p5.Vector {
+         var box = this.s.font_it.textBounds("x", 0, 1000, this.scale * this.s.baseFontSize);
+         return this.p.createVector(0, - box.h / 2);
+     }
+
+    constructor(p: any, s: any, state: string) {
+        super(p, s);
+        this.s = s;
+        this.stateString = state;
+        this.state = state;
+        switch (state) {
+            case 'aqueous':
+                this.state = '(aq)';
+                this.pythonSymbol = '(aq)';
+                this.mhchemSymbol = '(aq)'
+                this.latexSymbol = '\\text{(aq)}';
+                break;
+            case 'gas':
+                this.state = '(g)';
+                this.pythonSymbol = '(g)';
+                this.mhchemSymbol = '(g)'
+                this.latexSymbol = '\\text{(g)}';
+                break;
+            case 'solid':
+                this.state = '(s)';
+                this.pythonSymbol = '(s)';
+                this.mhchemSymbol = '(s)'
+                this.latexSymbol = '\\text{(s)}';
+                break;
+            case 'liquid':
+                this.state = '(l)';
+                this.pythonSymbol = '(l)';
+                this.mhchemSymbol = '(l)'
+                this.latexSymbol = '\\text{(l)}';
+                break;
+            case 'metal':
+                this.state = '(m)';
+                this.pythonSymbol = '(m)';
+                this.mhchemSymbol = '(m)'
+                this.latexSymbol = '\\text{(m)}';
+                break;
+            default:
+                this.state = state;
+                this.pythonSymbol = state;
+                this.latexSymbol = state;
+        }
+
+        // FIXME Not sure this is entirely right. Maybe make the "type" in DockingPoint an array? Works for now.
+        this.docksTo = ['chemical_element', "operator_brackets"];
+    }
+
+    /**
+     * Generates all the docking points in one go and stores them in this.dockingPoints.
+     * A Relation has one docking point:
+     *
+     - _right_: Symbol
+     */
+    generateDockingPoints() {
+        var box = this.boundingBox();
+        var descent = this.position.y - (box.y + box.h);
+
+        this.dockingPoints["right"] = new DockingPoint(this, this.p.createVector(box.w / 2 + this.s.mBox.w / 4, -this.s.xBox.h / 2), 1, "state_symbol", "right");
+    }
+
+    /**
+     * Generates the expression corresponding to this widget and its subtree.
+     *
+     * The `subscript` format is a special one for generating symbols that will work with the sympy checker. It squashes
+     * everything together, ignoring operations and all that jazz.
+     *
+     * @param format A string to specify the output format. Supports: latex, python, subscript.
+     * @returns {string} The expression in the specified format.
+     */
+    getExpression(format: string): string {
+        var expression = "";
+        if (format == "latex") {
+          expression = this.latexSymbol + "";
```

Comment:
Why do we need the ` + "";` in all these cases? Surely if it's an implicit typecast, it'd be much nicer to do `expression += this.insertSymbolTypeHere` ?